pkg/types/identifier.go

```diff
@@ -0,0 +1,48 @@
+package types
+
+import (
+	"strings"
+
+	purl "github.com/package-url/packageurl-go"
+)
+
+const (
+	PkgIdFormatCPE     = "cpe"
+	PkgIdFormatPURL    = "purl"
+	PkgIdFormatUnknown = "unknown"
+)
+
+// PkgIdentifier represents a software identifiers in any of the supported formats.
+type PkgIdentifier struct {
+	// Format is the software identifier format (e.g. CoSWID, CPE, PURL, etc.)
+	Format string
+	// Value represents the software identifier value
+	Value string
+}
```

Comment:
A single package can have multiple identifiers, like CPE and PURL. It should be able to accommodate all of them. One idea is make `PkgIdentifier` an array, but it is a bit bothering to access PURL because it needs to iterate the array every time. What if simply adding identifiers as the field?

```suggestion
type PkgIdentifier struct {
	PURL string `json:",omitempty"`
	CPE string    `json:",omitempty"`
}
```

Reply:
All right, I'll adopt this approach.